### ISSUE TITLE
[Bug]: /nb-events and notebooks API endpoints let any user to get data of any notebook

### ISSUE THREAD

**Issue by lucferbux** [opendatahub-io/odh-dashboard#570]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The API endpoint `nb-events` and `notebooks` should allow (no admin) users to get events from their notebook only. It works as expected if the API call points to a specific notebook, but if we leave the notebook  name empty, it returns the events from all the notebooks (which is an inconsistent behavior).
When pointing to a specific notebook which doesn't belong to the user (correct):
<img width="721" alt="Screenshot 2022-09-19 at 12 42 04" src="https://user-images.githubusercontent.com/16117276/191000602-908ffab8-df93-48a5-964a-efb37e34973d.png">
When not pointing to a specific notebook (wrong behavior):
<img width="795" alt="Screenshot 2022-09-19 at 12 42 23" src="https://user-images.githubusercontent.com/16117276/191000641-b5c50c5c-8695-482e-be32-a6a94d41176c.png">

### Steps To Reproduce

1. Log in RHODS Dashboard using any basic user "userA" (without admin privileges)
2. perform the following GET call: `<dashboard_url>/api/nb-events/rhods-notebooks/<notebook_userB>`
3. check the result
4. perform the following GET call: `<dashboard_url>/api/nb-events/rhods-notebooks/`

### Openshift Version

2.1.1

**Comment by andrewballantyne** [comment]
@lucferbux was this tested on main? Was it not fixed by https://github.com/opendatahub-io/odh-dashboard/pull/554?

Competing issue: https://github.com/opendatahub-io/odh-dashboard/issues/553

**Comment by andrewballantyne** [comment]
Ah, I think you were testing on a previous release... I saw an empty field in the description and you just added the info two fields up

```
Openshift Version
2.1.1

...

Open Data Hub Version
No response
```

**Comment by andrewballantyne** [comment]
Closing this issue as fixed by #553
